### PR TITLE
[Merged by Bors] - Make transactions in execution layer integration tests

### LINKED ISSUE #3159: Add transactions to EE integration tests
## Description

In https://github.com/sigp/lighthouse/pull/3157 we removed finalized payloads from the database in favour of reconstructing them from the execution layer. Our reconstruction depends on the correctness of the `ethers` library's transaction encoding, and it would be good to ensure this is correct in CI.

To do this we need to add some transactions to the execution blocks in our EE integration tests. This will involve creating and signing transactions, preferably of several different types (EIP-1559 and legacy), publishing them to the EE, ensuring they get included in payloads and that these payloads can be reconstructed.

Deeper testing could involve fuzzing, or live network testing (e.g. trying to reconstruct every block on Kiln)

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

Resolves #3159 

## Proposed Changes

Sends transactions to the EE before requesting for a payload in the `execution_integration_tests`. Made some changes to the integration tests in order to be able to sign and publish transactions to the EE:

1. `genesis.json` for both geth and nethermind was modified to include pre-funded accounts that we know private keys for 
2. Using the unauthenticated port again in order to make `eth_sendTransaction` and calls from the `personal` namespace to import keys

Also added a `fcu` call with `PayloadAttributes` before calling `getPayload` in order to give EEs sufficient time to pack transactions into the payload.

### CODE REVIEW

File: testing/execution_engine_integration/src/execution_engine.rs [hunk 1]
```diff
@@ -38,22 +49,31 @@ impl<E> Drop for ExecutionEngine<E> {
 impl<E: GenericExecutionEngine> ExecutionEngine<E> {
     pub fn new(engine: E) -> Self {
         let datadir = E::init_datadir();
+        // E::write_accounts(&datadir);
```
Comment (michaelsproul) [inline testing/execution_engine_integration/src/execution_engine.rs]: Can delete this

File: testing/execution_engine_integration/src/genesis_json.rs [hunk 2]
```diff
@@ -32,11 +32,102 @@ pub fn geth_genesis_json() -> Value {
...
+            "accounts": {
+              "0x7b8C3a386C0eea54693fFB0DA17373ffC9228139": {
+                "balance": "10000000000000000000000000"
+              },
+              "0xdA2DD7560DB7e212B945fC72cEB54B7D8C886D77": {
+                "balance": "10000000000000000000000000"
+              },
+            },
+            "nodes": [
+              "enode://c354db99124f0faf677ff0e75c3cbbd568b2febc186af664e0c51ac435609badedc67a18a63adb64dacc1780a28dcefebfc29b83fd1a3f4aa3c0eb161364cf94@164.92.130.5:30303"
```
Comment (michaelsproul) [inline testing/execution_engine_integration/src/genesis_json.rs]: Should we remove this so we're not pinging some random IP from CI?

File: testing/execution_engine_integration/src/transactions.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    transaction::{eip2718::TypedTransaction, eip2930::AccessList},
+    Address, Bytes, Eip1559TransactionRequest, TransactionRequest,
+};
+use types::{DepositData, EthSpec, Hash256, Keypair, Signature};
+
+/// Hardcoded deposit contract address based on sender address and nonce
+pub const DEPOSIT_CONTRACT_ADDRESS: &str = "64f43BEc7F86526686C931d65362bB8698872F90";
+
+#[derive(Debug)]
+pub enum Transactions {
```
Comment (michaelsproul) [inline testing/execution_engine_integration/src/transactions.rs]: The plural _transactions_ seems strange here, we could just use `Transaction`?

**Review by michaelsproul — CHANGES_REQUESTED** [review]
This is awesome Pawan, thanks!

I have just a few suggestions

File: testing/execution_engine_integration/src/genesis_json.rs (hunk 2 above)
Comment (pawanjay176) [reply testing/execution_engine_integration/src/genesis_json.rs]: Nice catch!

**Review by michaelsproul — APPROVED** [review]
bors r+

**Pull request closed without merging**